Target SPRacingF3NEO not building

From https://github.com/betaflight/betaflight/pull/8321#issue-280867950 reported by @hydra:

> when compiling master a linker error occurs due some baro code referenced somewhere (baroConfig_System undefined) but the defines and src definitions were missing.

The actual error I get is:

```
Linking SPRACINGF3NEO 
Memory region         Used Size  Region Size  %age Used
           FLASH:      246156 B       252 KB     95.39%
    FLASH_CONFIG:          0 GB         4 KB      0.00%
C:\Users\Hydra\AppData\Local\Temp\cculaMJF.ltrans0.ltrans.o: In function `findSerialPortIndexByIdentifier':
             RAM:       35800 B        40 KB     87.40%
             CCM:          2 KB         8 KB     25.00%
D:\My Documents\dev\embedded\betaflight/./src/main/io/serial.c:168: undefined reference to `barometerConfig_System'
       MEMORY_B1:          0 GB         0 GB     -1.#J%
collect2.exe: error: ld returned 1 exit status
make[1]: *** [Makefile:311: obj/main/betaflight_SPRACINGF3NEO.elf] Error 1
make[1]: Leaving directory '/cygdrive/d/My Documents/dev/embedded/betaflight'
make: *** [Makefile:472: hex] Error 2
"make TARGET=SPRACINGF3NEO -j12 DEBUG=INFO" terminated with exit code 2. Build might be incomplete.
```

Confirmed that it's not a build artifact as it still does it after issuing `rm -rf obj/main/SPRACINGF3NEO`

--- a/src/main/target/SPRACINGF3NEO/config.c
+++ b/src/main/target/SPRACINGF3NEO/config.c
@@ -23,33 +23,13 @@
 
 #include "platform.h"
 
-#include "common/axis.h"
-
-#include "drivers/sensor.h"
-#include "drivers/compass/compass.h"
 #include "drivers/serial.h"
 
-#include "fc/rc_controls.h"
-
-#include "flight/failsafe.h"
-#include "flight/mixer.h"
-#include "flight/pid.h"
-
 #include "pg/rx.h"
-
 #include "rx/rx.h"
 
 #include "io/serial.h"
-
 #include "telemetry/telemetry.h"
-
-#include "sensors/sensors.h"
-#include "sensors/compass.h"
-#include "sensors/barometer.h"
-
-#include "config/feature.h"
-
-#include "fc/config.h"
 
 #ifdef USE_TARGET_CONFIG
 
@@ -68,8 +48,6 @@ static targetSerialPortFunction_t targetSerialPortFunction[] = {
 
 void targetConfiguration(void)
 {
-    barometerConfigMutable()->baro_hardware = BARO_DEFAULT;
-    compassConfigMutable()->mag_hardware = MAG_DEFAULT;
     targetSerialPortFunctionConfig(targetSerialPortFunction, ARRAYLEN(targetSerialPortFunction));
     telemetryConfigMutable()->halfDuplex = true;
 }

